subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c

```diff
@@ -64,13 +65,14 @@ static inline CborError
 os_mgmt_taskstat_encode_thread_name(struct CborEncoder *encoder, int idx,
 				    const struct k_thread *thread)
 {
-	int err = 0;
+	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+	const int max_size = MIN(ARRAY_SIZE(thread_name), ARRAY_SIZE(thread->name)) - 1;
 
 	ARG_UNUSED(idx);
+	memcpy(thread_name, thread->name, max_size);
+	thread_name[max_size] = '\0';
 
-	err = cbor_encode_text_string(encoder, thread->name, strnlen(thread->name,
-				      CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1));
-	return err;
+	return cbor_encode_text_stringz(encoder, thread->name);
```

Comment:
I guess you want to pass `thread_name` here ?

Reply:
Fixed.